prow/cmd/build/controller.go

```diff
@@ -56,11 +57,13 @@ const (
 	controllerName = "prow-build-crd"
 )
 
+type getConfig func() *config.Config
+
 type controller struct {
-	pjNamespace string
-	pjc         prowjobset.Interface
-	builds      map[string]buildConfig
-	totURL      string
+	config getConfig
```

Comment:
What's the advantage compare with taking in the config agent? Do we want to unify the behavior across all our controllers?